go/vt/tabletserver/sqlquery.go

```diff
@@ -63,16 +62,20 @@ var stateName = []string{
 
 // SqlQuery implements the RPC interface for the query service.
 type SqlQuery struct {
-	// mu is used to manage state transitions.
-	// Obtain a write lock to change state.
-	// Obtain a read lock to prevent state from changing.
-	// If you want to know the current value of state and
-	// don't care that it changes after you've read it,
-	// you can perform a lockless atomic read.
-	mu       sync.RWMutex
+	// mu is used to access state. It's also used to ensure
+	// that state does not change out of SERVING or SHUTTING_TX
+	// while we do requests.Add.
+	// At the time of shut down, once we change the state to
+	// SHUTTING_QUERIES, no new requests will be honored.
+	// At this time, it's safe to perform requests.Wait outside
+	// the lock. Once the wait completes, we can transition
+	// to NOT_SERVING.
+	mu       sync.Mutex
+	state    int64
```

Comment:
I believe the state names cannot be trivially changed because of possible downstream dependencies, but I can change just the variable names, which will then not match the state names.